crates/context_aware_config/src/api/config/handlers.rs

```diff
@@ -147,6 +161,310 @@ async fn generate_cac(
+        let mut ct_dimensions = extract_dimensions(&context.condition).unwrap();
+        ct_dimensions.insert("key_val".to_string(), key_val);
+        let request_payload = json!({
+            "override": overrides,
+            "context": context.condition,
+            "id": context.id,
+            "to_be_deleted": overrides.is_empty(),
+            "override_id": override_id,
+        });
+        ct_dimensions.insert("req_payload".to_string(), request_payload);
+        dimensions.push(ct_dimensions);
+    }
+
+    //adding default config value
+    let mut default_config_map = Map::new();
+    if let Some(some_default_config_val) = default_config_val {
+        default_config_map
+            .insert("key_val".to_string(), some_default_config_val.to_owned());
+    };
+    dimensions.push(default_config_map);
+
+    for (index1, c1) in dimensions.clone().iter().enumerate() {
+        let mut temp_c1 = c1.clone();
+        let mut nc1 = c1.clone();
+        nc1.remove("req_payload");
+        let c1_val = nc1.remove("key_val");
+        let all_subsets_c1 = generate_subsets(&nc1);
+        let mut can_be_removed = false;
+        for (index2, c2) in dimensions.iter().enumerate() {
+            let mut temp_c2 = c2.clone();
+            temp_c2.remove("req_payload");
+            let c2_val = temp_c2.remove("key_val");
+            if index2 != index1 {
+                if all_subsets_c1.contains(&temp_c2) {
+                    if c1_val == c2_val {
+                        can_be_removed = true;
+                        break;
+                    } else if c2_val.is_some() {
+                        break;
+                    }
+                }
+            }
+        }
+        if can_be_removed {
+            if let Some(req_payload) = temp_c1.get("req_payload") {
+                let mut t = Map::new();
+                t.insert("req_payload".to_string(), req_payload.clone());
+                temp_c1 = t;
+            };
+        }
+        dimensions[index1] = temp_c1;
+    }
+    dimensions
+}
+
+fn get_contextids_from_overrideid(
+    contexts: Vec<Context>,
+    overrides: Map<String, Value>,
+    key_val: Value,
+    override_id: &str,
+) -> Vec<(Context, Map<String, Value>, Value, String)> {
+    let mut res: Vec<(Context, Map<String, Value>, Value, String)> = Vec::new();
+    for ct in contexts {
+        let ct_dimensions = extract_dimensions(&ct.condition).unwrap();
+        if ct_dimensions.contains_key("variantIds") {
+            continue;
+        }
+        let override_keys = &ct.override_with_keys;
+        if override_keys.contains(&override_id.to_owned()) {
+            res.push((
+                ct,
+                overrides.clone(),
+                key_val.clone(),
+                override_id.to_string(),
+            ));
+        }
+    }
+    res
+}
+
+fn construct_new_payload(req_payload: &Map<String, Value>) -> web::Json<PutReq> {
```

Comment:
Why not take a mutable cloned reference as an argument and skip the first line?

Reply:
Hey actually, i'll have to do the same where i'm calling also
Like i would have to clone it over there and send it.